examples/push-repo

```diff
@@ -0,0 +1,265 @@
+        "--release-date",
+        help="Date on which the content will be made available.",
+    )
+    parser.add_argument(
+        "--bucket", required=True, help="S3 bucket in which to upload file."
+    )
+    parser.add_argument(
+        "--table",
+        required=True,
+        help="DynamoDB table in which to publish file data.",
+    )
+    parser.add_argument(
+        "--aws-access-id",
+        default=None,
+        help="Access ID for Amazon services. If no ID is provided, attempts to"
+        " find it among environment variables and ~/.aws/config file will"
+        " be made",
+    )
+    parser.add_argument(
+        "--aws-access-key",
+        default=None,
+        help="Access key for Amazon services. If no key is provided, attempts"
+        " to find it among environment variables and ~/.aws/config file"
+        " will be made",
+    )
+    parser.add_argument(
+        "--aws-session-token",
+        default=None,
+        help="Session token for Amazon services. If no token is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+    parser.add_argument(
+        "--default-region",
+        default=None,
+        help="Default region for Amamzon services. If no region is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+    parser.add_argument(
+        "--dryrun",
+        action="store_true",
+        help="Don't execute the action, only log what would otherwise be done.",
+    )
+    parser.add_argument(
+        "--debug", action="store_true", help="Include debug logging."
+    )
+
+    p = parser.parse_args()
+
+    if p.debug:
+        logging.getLogger("chexus").setLevel(logging.DEBUG)
+        LOG.setLevel(logging.DEBUG)
+
+    client = Client(
+        access_id=p.aws_access_id,
+        access_key=p.aws_access_key,
+        session_token=p.aws_session_token,
+        default_region=p.default_region,
+    )
+
+    is_iso = True if p.repo_type == "iso-repo" else False
+
+    for root_dir, subdir_list, file_list in os.walk(p.local_repo):
+        for file_name in file_list:
+            file_path = os.path.join(root_dir, file_name)
+            upl_item = BucketItem(file_path)
+            upl_item.key = upl_item.checksum
+
+            origin_item = TableItem(
+                object_key=upl_item.checksum,
+                web_uri=cdn_path(upl_item.name, file_path, upl_item.checksum),
+                from_date=p.release_date,
+                metadata=p.file_metadata,
+            )
+            print(file_path)
+            print(root_dir)
+            repo_item = TableItem(
+                object_key=upl_item.checksum,
+                web_uri=repo_path(
+                    p.local_repo, file_name, root_dir, p.dest_repo, is_iso
+                ),
+                from_date=p.release_date,
+                metadata=p.file_metadata,
+            )
```

Comment:
For setting the web_uri, I think the appropriate behavior here is more like:

- relative URI under remote repo root is the file's relative path under the local repo root
- repo_path isn't required
- is_iso isn't required

So that would be something like:

```suggestion
            rel_file_path = os.path.relpath(file_path, p.local_repo)
            repo_item = TableItem(
                object_key=upl_item.checksum,
                web_uri=os.path.join(p.dest_repo, rel_file_path),
                from_date=p.release_date,
                metadata=p.file_metadata,
            )
```

It uses exactly the same directory structure in the remote repo as used locally, which seems correct in all cases without having to know what kind of content we're dealing with.  I can't see any reason why we need to add cases for different content types.

Reply:
> I can't see any reason why we need to add cases for different content types.

Nor can I. I'm not sure where I was going with that. Thanks for the suggestion!